src/transformers/modeling_outputs.py

```diff
@@ -95,6 +112,7 @@ class BaseModelOutputWithPooling(ModelOutput):
     pooler_output: torch.FloatTensor = None
     hidden_states: Optional[Tuple[torch.FloatTensor]] = None
     attentions: Optional[Tuple[torch.FloatTensor]] = None
+    intermediate_activations: Optional[Tuple[torch.FloatTensor]] = None
```

Comment:
Hmm this should definitely not be done, is this a mistake?

Reply:
Okay makes sense, but I would rather put it on `modeling_output.py` since we might need that for future models too

Reply:
I wouldn't be surprised to see a vision model in the future that uses intermediate activations from the backbone or anywhere else from the model

Reply:
Proposed something in [1a39856](https://github.com/huggingface/transformers/pull/20645/commits/1a398563877fde6bdfd69d1d4db3d15cb0ba7702) & [63bf08a](https://github.com/huggingface/transformers/pull/20645/commits/63bf08a0cc2b12e142105f86cc2999bd3382151e)

Reply:
Finally moved the dataclasses in 275f25c